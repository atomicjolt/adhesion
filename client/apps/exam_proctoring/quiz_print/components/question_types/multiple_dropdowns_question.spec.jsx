import React from 'react';
import { shallow } from 'enzyme';
import MultipleDropdown from './multiple_dropdowns_question';

describe('Multiple DropDown', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      answers: [
        {
          blank_id: 'Skyrim',
          id: 1,
          text: 'Bretons'
        },
        {
          blank_id: 'Skyrim',
          id: 2,
          text: 'Nords'
        },
        {
          blank_id: 'Skyrim',
          id: 3,
          text: 'Argonians'
        },
        {
          blank_id: 'Marrowind',
          id: 4,
          text: 'Dunmer'
        },
        {
          blank_id: 'Marrowind',
          id: 5,
          text: 'Nords'
        },
        {
          blank_id: 'Marrowind',
          id: 6,
          text: 'Falmer'
        },
      ],
      question_text: '<p>Skyrim belongs to the [Skyrim], while Marrowind is the homeland of the [Marrowind].</p>',
      id: '7',
    };
    result = shallow(<MultipleDropdown {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('renders radio buttons for the options', () => {
    const radios = result.find('input');
    expect(radios.length).toBe(6);
    expect(radios.at(0).props().type).toBe('radio');
  });
});
