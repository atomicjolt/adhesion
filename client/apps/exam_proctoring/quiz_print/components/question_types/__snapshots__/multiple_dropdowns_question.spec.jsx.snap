// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple DropDown matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <multipleDropdowns
    answers={
        Array [
            Object {
              "blank_id": "Skyrim",
              "id": 1,
              "text": "Bretons",
            },
            Object {
              "blank_id": "Skyrim",
              "id": 2,
              "text": "Nords",
            },
            Object {
              "blank_id": "Skyrim",
              "id": 3,
              "text": "Argonians",
            },
            Object {
              "blank_id": "Marrowind",
              "id": 4,
              "text": "Dunmer",
            },
            Object {
              "blank_id": "Marrowind",
              "id": 5,
              "text": "Nords",
            },
            Object {
              "blank_id": "Marrowind",
              "id": 6,
              "text": "Falmer",
            },
          ]
    }
    id="7"
    question_text="<p>Skyrim belongs to the [Skyrim], while Marrowind is the homeland of the [Marrowind].</p>"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <essayQuestion
          text="<p>Skyrim belongs to the [Skyrim], while Marrowind is the homeland of the [Marrowind].</p>"
/>,
        Array [
          <div
            style={
                        Object {
                                    "border": "1px solid darkGrey",
                                    "display": "inline-block",
                                    "margin": "5px",
                                    "padding": "5px",
                                    "verticalAlign": "top",
                                  }
            }
>
            <strong>
                        Skyrim
            </strong>
            <hr />
            <div>
                        <input
                                    disabled={true}
                                    type="radio"
                        />
                         
                        Bretons
            </div>
            <div>
                        <input
                                    disabled={true}
                                    type="radio"
                        />
                         
                        Nords
            </div>
            <div>
                        <input
                                    disabled={true}
                                    type="radio"
                        />
                         
                        Argonians
            </div>
</div>,
          <div
            style={
                        Object {
                                    "border": "1px solid darkGrey",
                                    "display": "inline-block",
                                    "margin": "5px",
                                    "padding": "5px",
                                    "verticalAlign": "top",
                                  }
            }
>
            <strong>
                        Marrowind
            </strong>
            <hr />
            <div>
                        <input
                                    disabled={true}
                                    type="radio"
                        />
                         
                        Dunmer
            </div>
            <div>
                        <input
                                    disabled={true}
                                    type="radio"
                        />
                         
                        Nords
            </div>
            <div>
                        <input
                                    disabled={true}
                                    type="radio"
                        />
                         
                        Falmer
            </div>
</div>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "text": "<p>Skyrim belongs to the [Skyrim], while Marrowind is the homeland of the [Marrowind].</p>",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "answer_bank_Skyrim",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Skyrim
</strong>,
            <hr />,
            Array [
              <div>
                <input
                                disabled={true}
                                type="radio"
                />
                 
                Bretons
</div>,
              <div>
                <input
                                disabled={true}
                                type="radio"
                />
                 
                Nords
</div>,
              <div>
                <input
                                disabled={true}
                                type="radio"
                />
                 
                Argonians
</div>,
            ],
          ],
          "style": Object {
            "border": "1px solid darkGrey",
            "display": "inline-block",
            "margin": "5px",
            "padding": "5px",
            "verticalAlign": "top",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Skyrim",
            },
            "ref": null,
            "rendered": "Skyrim",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": "dropdown_answer_1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  disabled={true}
                  type="radio"
/>,
                " ",
                "Bretons",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                  "type": "radio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              " ",
              "Bretons",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "dropdown_answer_2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  disabled={true}
                  type="radio"
/>,
                " ",
                "Nords",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                  "type": "radio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              " ",
              "Nords",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "dropdown_answer_3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  disabled={true}
                  type="radio"
/>,
                " ",
                "Argonians",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                  "type": "radio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              " ",
              "Argonians",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "answer_bank_Marrowind",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Marrowind
</strong>,
            <hr />,
            Array [
              <div>
                <input
                                disabled={true}
                                type="radio"
                />
                 
                Dunmer
</div>,
              <div>
                <input
                                disabled={true}
                                type="radio"
                />
                 
                Nords
</div>,
              <div>
                <input
                                disabled={true}
                                type="radio"
                />
                 
                Falmer
</div>,
            ],
          ],
          "style": Object {
            "border": "1px solid darkGrey",
            "display": "inline-block",
            "margin": "5px",
            "padding": "5px",
            "verticalAlign": "top",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Marrowind",
            },
            "ref": null,
            "rendered": "Marrowind",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": "dropdown_answer_4",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  disabled={true}
                  type="radio"
/>,
                " ",
                "Dunmer",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                  "type": "radio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              " ",
              "Dunmer",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "dropdown_answer_5",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  disabled={true}
                  type="radio"
/>,
                " ",
                "Nords",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                  "type": "radio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              " ",
              "Nords",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "dropdown_answer_6",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  disabled={true}
                  type="radio"
/>,
                " ",
                "Falmer",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                  "type": "radio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              " ",
              "Falmer",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <essayQuestion
            text="<p>Skyrim belongs to the [Skyrim], while Marrowind is the homeland of the [Marrowind].</p>"
/>,
          Array [
            <div
              style={
                            Object {
                                          "border": "1px solid darkGrey",
                                          "display": "inline-block",
                                          "margin": "5px",
                                          "padding": "5px",
                                          "verticalAlign": "top",
                                        }
              }
>
              <strong>
                            Skyrim
              </strong>
              <hr />
              <div>
                            <input
                                          disabled={true}
                                          type="radio"
                            />
                             
                            Bretons
              </div>
              <div>
                            <input
                                          disabled={true}
                                          type="radio"
                            />
                             
                            Nords
              </div>
              <div>
                            <input
                                          disabled={true}
                                          type="radio"
                            />
                             
                            Argonians
              </div>
</div>,
            <div
              style={
                            Object {
                                          "border": "1px solid darkGrey",
                                          "display": "inline-block",
                                          "margin": "5px",
                                          "padding": "5px",
                                          "verticalAlign": "top",
                                        }
              }
>
              <strong>
                            Marrowind
              </strong>
              <hr />
              <div>
                            <input
                                          disabled={true}
                                          type="radio"
                            />
                             
                            Dunmer
              </div>
              <div>
                            <input
                                          disabled={true}
                                          type="radio"
                            />
                             
                            Nords
              </div>
              <div>
                            <input
                                          disabled={true}
                                          type="radio"
                            />
                             
                            Falmer
              </div>
</div>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": "<p>Skyrim belongs to the [Skyrim], while Marrowind is the homeland of the [Marrowind].</p>",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "answer_bank_Skyrim",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Skyrim
</strong>,
              <hr />,
              Array [
                <div>
                  <input
                                    disabled={true}
                                    type="radio"
                  />
                   
                  Bretons
</div>,
                <div>
                  <input
                                    disabled={true}
                                    type="radio"
                  />
                   
                  Nords
</div>,
                <div>
                  <input
                                    disabled={true}
                                    type="radio"
                  />
                   
                  Argonians
</div>,
              ],
            ],
            "style": Object {
              "border": "1px solid darkGrey",
              "display": "inline-block",
              "margin": "5px",
              "padding": "5px",
              "verticalAlign": "top",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Skyrim",
              },
              "ref": null,
              "rendered": "Skyrim",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": "dropdown_answer_1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    disabled={true}
                    type="radio"
/>,
                  " ",
                  "Bretons",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "disabled": true,
                    "type": "radio",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                " ",
                "Bretons",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "dropdown_answer_2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    disabled={true}
                    type="radio"
/>,
                  " ",
                  "Nords",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "disabled": true,
                    "type": "radio",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                " ",
                "Nords",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "dropdown_answer_3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    disabled={true}
                    type="radio"
/>,
                  " ",
                  "Argonians",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "disabled": true,
                    "type": "radio",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                " ",
                "Argonians",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "answer_bank_Marrowind",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Marrowind
</strong>,
              <hr />,
              Array [
                <div>
                  <input
                                    disabled={true}
                                    type="radio"
                  />
                   
                  Dunmer
</div>,
                <div>
                  <input
                                    disabled={true}
                                    type="radio"
                  />
                   
                  Nords
</div>,
                <div>
                  <input
                                    disabled={true}
                                    type="radio"
                  />
                   
                  Falmer
</div>,
              ],
            ],
            "style": Object {
              "border": "1px solid darkGrey",
              "display": "inline-block",
              "margin": "5px",
              "padding": "5px",
              "verticalAlign": "top",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Marrowind",
              },
              "ref": null,
              "rendered": "Marrowind",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": "dropdown_answer_4",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    disabled={true}
                    type="radio"
/>,
                  " ",
                  "Dunmer",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "disabled": true,
                    "type": "radio",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                " ",
                "Dunmer",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "dropdown_answer_5",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    disabled={true}
                    type="radio"
/>,
                  " ",
                  "Nords",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "disabled": true,
                    "type": "radio",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                " ",
                "Nords",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "dropdown_answer_6",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    disabled={true}
                    type="radio"
/>,
                  " ",
                  "Falmer",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "disabled": true,
                    "type": "radio",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                " ",
                "Falmer",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
